src/transaction/components/TransactionProvider.tsx

```diff
@@ -104,6 +112,11 @@ export function TransactionProvider({
         } catch (_err) {
           setErrorMessage(genericErrorMessage);
         }
+      } else if (
```

Comment:
Can we add a line of comment for those conditions, so we don't forget for what are for.